framework/base/Component.php

```diff
@@ -475,12 +474,18 @@ public function hasEventHandlers($name)
      */
     public function on($name, $handler, $data = null, $append = true)
     {
+        $priority = (int)$append;
         $this->ensureBehaviors();
-        if ($append || empty($this->_events[$name])) {
-            $this->_events[$name][] = [$handler, $data];
-        } else {
-            array_unshift($this->_events[$name], [$handler, $data]);
+
+        if (empty($this->_events[$name])) {
+            $this->_events[$name] = \Yii::createObject('yii\base\EventPriorityQueue');
         }
+
+        if (!$priority) {
+            $priority = $this->_events[$name]->getMaxPriority() + 5;
```

Comment:
5 is used until is deprecated because of the current append prepend functionality. When priorities in core flow defined we 'll just remove that 5.